i18n/detect.py

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env python
+"""
+Function to check whether or not the English strings are up to date
+"""
+from __future__ import print_function
+from subprocess import CalledProcessError
+
+from i18n import Runner
+from i18n.execute import execute
+from i18n.extract import Extract
+
+
+class Detect(Runner):
+    """
+    Class used to check if the English source strings are up to date
+    """
+    def run(self, args):
+        """
+        Main entry point of script
+        """
+        Extract().run(args)  # Run extract command
+        changes = self.detect_changes()
+        message = self.get_message(changes)
+        print(message)
+
+    def detect_changes(self):
+        """
+        Detect if extract command changed msgid or msgstr strings.
+        Will return 1 if changes detected and 0 when no changes detected.
+        """
+        try:
+            execute('git diff --exit-code -G "^[msgid|msgstr]"')
+        except CalledProcessError as ex:
+            if ex.returncode == 1:
+                return 1
+        return 0
+
+    def get_message(self, changes):
+        """
+        Returns message depending of argument changes
+        """
+        changes_detected_message = 'Changes detected! Please create the source files and push them to GitHub.'
```

Comment:
Again, no dependency on GitHub. Also, this can be made somewhat easier to read:

```python
msg = 'Source translation files are current.'

if changes_detected:
  msg = 'Source translations are out-of-date! Please update them.'

return msg
```